WIP PR2 for BXMSDOC-1225: More details about user permission customization and security management.

kaldesai opened the pull request:
**JIRA:** 
- [Dedicated JIRA](https://issues.redhat.com/browse/BXMSDOC-1225)

**Enterprise doc previews:**
- [RHPAM-Configuring Business Central settings and properties](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-6599-RHPAM-SM-PeerReview/#business-central-security-management-con_configuring-central)
- [RHDM-Configuring Business Central settings and properties](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-6599-RHDM-SM/#business-central-security-management-con_configuring-central)

**Community doc preview:**
- [jBPM community document](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-6599-SM-jBPM/#business-central-security-management-con_business-processes)
- [Drools community document(same as jBPM)](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-6599-SM-Drools/#business-central-security-management-con_kie-apis)

**Doc impact:** 
Please check the following chapter from **RHPAM** doc:
26. Security management

Please check the following chapter from **RHDM** doc:
20. Security management

Please check the following chapter from **jBPM community** doc:
11.8. User and group management
11.9. Security management

heathjoy commented on doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-dashboard-proc.adoc:
```diff
@@ -8,7 +8,7 @@ There are four stages in the creation of a custom dashboard:
 * Data set authoring - In this stage, you define a data set for accessing the data and displaying it through the pages. For more information, see xref:data_sets_add_proc_building-custom-dashboard-widgets[Adding data sets].
 * Page authoring - In this stage, you create the pages which are used to display the data from the data sets, among other things. For more information, see xref:building-custom-dashboard-widgets-creating-pages-proc[Creating pages].
 * Publication - In this stage, navigation between pages are defined when you create the custom navigation trees or modify the existing default one (*Workbench* tree). For more information, see xref:building-custom-dashboard-widgets-creating-navigation-tree-proc[Creating a navigation tree] or xref:building-custom-dashboard-widgets-editing-navigation-tree-con[Editing a navigation tree].
-* Security management - In this stage, role and group permissions are set which defines the privileges a user enjoys when working on {CENTRAL}. For more information, see xref:business-central-settings-security-proc_building-custom-dashboard-widgets[Security management].
```
Consider changing the word "enjoys" to "in granted"

kaldesai replied:
Done

heathjoy commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-permissions-con.adoc:
```diff
@@ -0,0 +1,12 @@
+
+= Permissions
+
+A permission is an authorization granted to the user to do some actions related to a specific resource within the application. For example, user can have following permissions:
```
change "to do some actions" to "perform actions"

kaldesai replied:
Done

RishiRajAnand commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-provider-con.adoc:
```diff
@@ -0,0 +1,51 @@
+
+* Tomcat security management provider
++
+This provider supports realm types based on `XML` files. The default value is tomcat server, such as `tomcat-users.xml`. The classes used in this package contain the user and group manager implementations for Apache Tomcat.
```
@kaldesai Instead of "The default value is tomcat server, such as `tomcat-users.xml`", we can replace it with something like "The default file in tomcat server is tomcat-users.xml".

RishiRajAnand commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-changing-permissions-proc.adoc:
```diff
@@ -1,15 +1,23 @@
+image::Workbench/SecurityManagement/set-permissions.png[Setting the permissions]
++
 You can expand the resource type whose permissions you want to change.
 . Optional: To add an item not listed, click *Add Exception* and then set the permissions as required.
```
@kaldesai I think "To add an item not listed" conveys a contrasting message, lets replace with something like "To add an exception for a resource type".

kaldesai replied:
Done

tomasdavidorg commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-configuring-EAP-security-management-provider-proc.adoc:
```diff
@@ -0,0 +1,67 @@
+[id='business-central-configuring-EAP-security-management-provider-proc']
+
+= Configuring Wildfly or {EAP} security management provider
```
We don't support Wildfly. Only the EAP server is supported. Could you please modify the document to contain only EAP?

kaldesai replied:
Okay. 
I will remove the mentions about Wildefly.

kaldesai replied:
Done

tomasdavidorg commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-provider-con.adoc:
```diff
@@ -0,0 +1,51 @@
+
+|===
+
+* Tomcat security management provider
```
We don't support deploying Business central to Tomcat server. I think this part should be removed

kaldesai replied:
I have removed this section.

tomasdavidorg commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-configuring-tomcat-security-management-provider-proc.adoc:
```diff
@@ -0,0 +1,30 @@
+[id='business-central-configuring-tomcat-security-management-provider-proc']
```
We don't support deploying Business central to Tomcat server. I think this part should be removed

tomasdavidorg commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-enabling-security-management-proc.adoc:
```diff
@@ -0,0 +1,17 @@
+[id='business-central-enabling-security-management-proc']
+
+= Enabling the security management feature
```
@RishiRajAnand The security management feature is enabled by default, isn't it? Do we need those steps in the documentation for the deployed Business Central?

tomasdavidorg replied:
@kaldesai Could you please remove this file as it is not used anymore?

kaldesai replied:
@tomasdavidorg, I have removed this file.

tomasdavidorg commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-disabling-security-management-proc.adoc:
```diff
@@ -0,0 +1,10 @@
+[id='business-central-disabling-security-management-proc']
```
@RishiRajAnand Same question as for the `Enabling the security management feature`. Do we need this chapter? Is there any use case where a user wants to create own security implementation?

tomasdavidorg replied:
@kaldesai Could you please remove this file as it is not used anymore?

kaldesai replied:
@tomasdavidorg, I have removed this file.

tomasdavidorg commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-configuring-EAP-security-management-provider-proc.adoc:
```diff
@@ -0,0 +1,67 @@
+* Wildfly or {EAP} is installed on the system.
+* If you want to use an existing users or roles property files from the Wildfly or {EAP} instance, you must include the following system properties:
++
+.Wildfly or {EAP} security management provider based on property files
```
@RishiRajAnand do we have any use case where user wants to use its own implementation of security management provider? If the business central integrated with RH-SSO, we have special documentation for that.

tomasdavidorg commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-security-policy-con.adoc:
```diff
@@ -0,0 +1,25 @@
+[id='business-central-settings-security-policy-con']
+
+= Security policy
```
@RishiRajAnand Isn't this just default policy which is applied on the first start of Business Central and then it is stored in the NIOGIT and modified by the user interface? Maybe we should provide steps how to set up default configuration for business central. Wdyt?

tomasdavidorg commented:
@kaldesai @RishiRajAnand I am not sure what are the use cases for this documentation. I would say that user should not to modify security management providers and change security policy inside the NioGit directory. Maybe I am wrong but I think user suppose to use the UI for the security management configuration and maybe configure the default policy for the roles.

kaldesai commented:
@tomasdavidorg @RishiRajAnand, as per our discussion today I have updated the doc content. Can you please review and verify the content? I have updated the links as well. Thank you so much.

tomasdavidorg commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-configuring-tomcat-security-management-provider-proc.adoc:
```diff
@@ -0,0 +1,30 @@
+[id='business-central-configuring-tomcat-security-management-provider-proc']
```
Could we remove this file?

kaldesai replied:
Removed the tomcat provider file.

tomasdavidorg commented on doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-dashboard-proc.adoc:
```diff
@@ -8,7 +8,7 @@ There are four stages in the creation of a custom dashboard:
 * Page authoring - In this stage, you create the pages which are used to display the data from the data sets, among other things. For more information, see xref:building-custom-dashboard-widgets-creating-pages-proc[Creating pages].
 * Publication - In this stage, navigation between pages are defined when you create the custom navigation trees or modify the existing default one (*Workbench* tree). For more information, see xref:building-custom-dashboard-widgets-creating-navigation-tree-proc[Creating a navigation tree] or xref:building-custom-dashboard-widgets-editing-navigation-tree-con[Editing a navigation tree].
-* Security management - In this stage, role and group permissions are set which defines the privileges a user enjoys when working on {CENTRAL}. For more information, see xref:business-central-settings-security-proc_building-custom-dashboard-widgets[Security management].
+//* Security management - In this stage, role and group permissions are set which defines the privileges a user enjoys when working on {CENTRAL}. For more information, see xref:business-central-settings-security-proc_building-custom-dashboard-widgets[Security management].
```
Is this commented on purpose?

kaldesai replied:
Yes, need to update the Xref URL. I have removed the URL for time being and uncommented the information.

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-security-policy-con.adoc:
```diff
@@ -0,0 +1,25 @@
+
+Every application contains a single security policy which is used every time the system checks a permission. In security policy, the set of permissions are assigned to every role and a group.
+
+The security policy file is stored in the `WEB-INF/classes/security-policy.properties` under your application's `WAR` file structure.
```
Done

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-security-policy-con.adoc:
```diff
@@ -0,0 +1,25 @@
+role.user.permission.perspective.read.Dashboard=true
+----
+
+Every entry defines a single permission which is assigned to a role or group. When an application starts, the security policy file is loaded and stored in the memory.
```
Done.

tomasdavidorg reviewed (approved):
Thanks. Looks good to me.

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-provider-con.adoc:
```diff
@@ -0,0 +1,31 @@
+
+= Security management providers
+
+A security environment is provided by the use of a realm. Realms are used to restrict access to the different application resources. These realms contains the information about the users, groups, roles, and permissions. A security provider is the formal name given to a concrete user and group management service implementation for a given realm.
```
It is a security management provider and I have updated it in the content.

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-provider-con.adoc:
```diff
@@ -0,0 +1,31 @@
+
+The user & group management features available will depend on the security provider configured. If the built-in providers do not fit with your application security realm, then you can easily build and register your own security management provider.
+
+NOTE: If the security management provider is not installed, the user interface for managing the security realm will not be available. Once a management security provider is installed and setup, the user and group management features are automatically enabled in the security management user interface.
```
Done.

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-provider-con.adoc:
```diff
@@ -0,0 +1,31 @@
++
+This security provider supports realm types based on `application-users.properties` or `application-roles.properties` property files.
++
+You can use the following system properties for the {EAP} security management provider implementation:
```
Done

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-provider-con.adoc:
```diff
@@ -0,0 +1,31 @@
+
+NOTE: If the security management provider is not installed, the user interface for managing the security realm will not be available. Once a management security provider is installed and setup, the user and group management features are automatically enabled in the security management user interface.
+
+{CENTRAL} provides the following security management provider:
```
Done

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-configuring-EAP-security-management-provider-proc.adoc:
```diff
@@ -0,0 +1,67 @@
+|===
+
++
+.{EAP} security management provider based on property files and CLI mode
```
There are two ways to configure this EAP security management provider: 
- based on property files only
- based on property files and CLI mode

and system properties are different for both of them that is why I included two tables.

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-permissions-con.adoc:
```diff
@@ -0,0 +1,12 @@
+[id='business-central-security-management-permissions-con']
```
Done

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-changing-permissions-proc.adoc:
```diff
@@ -1,26 +1,29 @@
-= Changing permissions for groups and roles
+= Changing permissions for groups and roles in {CENTRAL}
 
 You cannot change permissions for an individual user. However, you can change permissions for groups and roles. The changed permissions apply to users that have the role or belong to the group that you changed.
```
Done

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-permissions-con.adoc:
```diff
@@ -0,0 +1,12 @@
+
+= Permissions and settings
+
+A permission is an authorization granted to the user to perform actions related to a specific resource within the application. For example, user can have following permissions:
```
Updated.

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-configuring-EAP-security-management-provider-proc.adoc:
```diff
@@ -0,0 +1,67 @@
+.Prerequisites
+
+* {EAP} is installed on the system.
+* If you want to use an existing users or roles property files from the {EAP} instance, you must include the following system properties:
```
We can include the system properties in these files:
- EAP_HOME/standalone/configuration/application-users.properties
- EAP_HOME/standalone/configuration/application-roles.properties

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-configuring-EAP-security-management-provider-proc.adoc:
```diff
@@ -0,0 +1,67 @@
+To configure {EAP} Security management provider do any of the following tasks:
+
+--
+If you want to use {EAP} security provider based on property files, do the following tasks:
```
If I convert it into a bullet then the steps which are present inside this and not listed in the numbered format. Even they are rendering as bullets.

kaldesai commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-con.adoc:
```diff
@@ -0,0 +1,5 @@
+
+= Security management
+
+You can perform security and permission management tasks from the *Security management* page. To access the *Security management* page, select the *Admin* icon in the top-right corner of the screen and click *Roles*, *Groups*, or *Users* on the {CENTRAL} *Settings* page. Depending on which icon you click, the *Security management* page opens for the respective tab. Use roles and groups to manage  ...
```
I have created a separate concept and procedure module for this content.

emmurphy1 reviewed (commented):
Kalyani, please have a look at this section as a whole and then try to update the first concept module to reflect the content of the section. I have made some suggestions. Also, think about the order of the modules. Does it matter? I don't know why I need to know about the Security Policy. Do you tell me how to define it? It just doesn't make sense. Seems like sections are missing. Does it make sense to you?

kaldesai commented:
Hey @emmurphy1, 

I added your suggestions in the content. 

1. Taking about security management introduction section I could not think about anything else because whatever I have added in the intro part explains everything about it and what it actually does. 
Can you please guide me on that? 

2. Also, I checked the security policy section and although it is good to have we don't need to have a separate section about it so I have removed it from the. I previously added this section just to let us know where the security policy is stored. 
Regarding security policy,  this happens automatically when I set permissions in business central. 

3. Removed unnecessary fluff **accessing-security-management** procedure and managed those steps in the last changing permission procedure sections. 

Can you please review and verify the content? 

Thank you.

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-con.adoc:
```diff
@@ -0,0 +1,7 @@
+
+= Security management
+
+Security management describes how you can manage the users, groups, and permissions with the help of user interface in order to configure who can access the different available resources and features.
```
Security management is the process of managing users, groups, and permissions.  
Delete this
 with the help of user interface in order to configure who can access the different available resources and features.

Rethink this paragraph. There is repetition. Try consolidate it.

kaldesai replied:
I have removed the repetitive content.

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-changing-permissions-proc.adoc:
```diff
@@ -0,0 +1,31 @@
+. To access the *Security management* page in {CENTRAL}, select the *Admin* icon in the top-right corner of the screen.
+. Click *Roles*, *Groups*, or *Users* on the {CENTRAL} *Settings* page.
++
+Depending on which icon you click, the *Security management* page opens for the respective tab.
```
The *Security management* page opens on the tab for the icon that you clicked.

kaldesai replied:
Done

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-configuring-EAP-security-management-provider-proc.adoc:
```diff
@@ -0,0 +1,67 @@
+[id='business-central-configuring-EAP-security-management-provider-proc']
+
+= Configuring {EAP} security management provider
+
```
I still don't see a reason for this procedure. Something like:
You can configure the application-users.properties or application-roles.properties property file.(to do what?)
In this section you are using 'security provider' but elsewhere 'security management provider'. Use one term consistently.
You are missing articles:
To use THE Red Hat JBoss EAP security provider... search for all instances.

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-configuring-EAP-security-management-provider-proc.adoc:
```diff
@@ -0,0 +1,72 @@
+
+.Procedure
+
+To configure {EAP} Security management provider, do any of the following tasks:
```
complete, not 'do'. Search for.
Also, the nested numbered list inside the bulleted list isn't working. I think you will have to make two procedures. So this section will be a concept module followed by two procedures.


[source]
----
`src/main/resources/security-management.properties`
----
----
Remove the ticks when you wrap inside source./

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-permissions-con.adoc:
```diff
@@ -0,0 +1,12 @@
+
+A permission is an authorization granted to a user to perform actions related to a specific resource within the application. For example, a user can have following permissions:
+
+* View a page
```
Need full stops after these.

kaldesai replied:
Done

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-permissions-con.adoc:
```diff
@@ -0,0 +1,12 @@
+* View a repository
+* Delete a dashboard
+
+A permission can be granted or denied and it can be global or resource specific. You can use permissions to protect access to resources and customize features within the application.
```
You can grant or deny permission and a permission can be global or resource specific.

kaldesai replied:
Updated

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-provider-con.adoc:
```diff
@@ -0,0 +1,9 @@
+
+In the context of security management, a realm restricts access to different application resources. Realms contain information about the users, groups, roles, and permissions. A concrete user and group management service implementation for a specific realm is called a security management provider.
+
+If the built-in security management providers do not fit with your application security realm, then you can build and register your own security management provider.
```
do not fit with your application security realm, change to:
do not meet the requirements of your application security realm

kaldesai replied:
Updated

kaldesai commented on doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-dashboard-proc.adoc:
```diff
@@ -8,7 +8,7 @@ There are four stages in the creation of a custom dashboard:
 * Page authoring - In this stage, you create the pages which are used to display the data from the data sets, among other things. For more information, see xref:building-custom-dashboard-widgets-creating-pages-proc[Creating pages].
 * Publication - In this stage, navigation between pages are defined when you create the custom navigation trees or modify the existing default one (*Workbench* tree). For more information, see xref:building-custom-dashboard-widgets-creating-navigation-tree-proc[Creating a navigation tree] or xref:building-custom-dashboard-widgets-editing-navigation-tree-con[Editing a navigation tree].
-* Security management - In this stage, role and group permissions are set which defines the privileges a user enjoys when working on {CENTRAL}. For more information, see xref:business-central-settings-security-proc_building-custom-dashboard-widgets[Security management].
+* Security management - In this stage, role and group permissions are set which defines the privileges are granted to a user when working on {CENTRAL}.
```
Added

kaldesai commented:
Hey @emmurphy1, 

I have incorporated your following suggestions in the content: 

- Added `security management provider` term instead of a `security provider` to maintain consistency.
- Included the missing articles.
- Added missing full stops.
- Because the nested numbered list inside the bulleted list isn't working I converted it into two procedures: 
Can titles be this long? 
Configuring the Red Hat JBoss EAP security management provider based on property files
Configuring the Red Hat JBoss EAP security management provider based on property files and CLI mode

Can you please review and verify the content of the Security management section? I have updated the document preview links. 

Also, the answer to the question you asked: 
**Question:** Why do we need to configure the security management providers?
**Answer:** Business central as an application can be deployed on various platforms like Jboss EAP, Tomcat, keycloak and so on. Right now only EAP is supported. 
So based on the target platform we need to configure a respective provider to provide it with the security features.
This is because every platform has its own way of managing security. That is also the reason why its called a "provider"
because it "provides" security management features for that particular platform. 

Please reach out to me if you have any concern or question. Thank you.

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-con.adoc:
```diff
@@ -0,0 +1,7 @@
+
+= Security management
+
+Security management is the process of managing users, groups, and permissions. You can control the access to {CENTRAL} resources and features from the {CENTRAL} Security management page.
```
You can control access to {CENTRAL} resources

kaldesai replied:
Done.

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-con.adoc:
```diff
@@ -0,0 +1,7 @@
+
+Security management is the process of managing users, groups, and permissions. You can control the access to {CENTRAL} resources and features from the {CENTRAL} Security management page.
+
+{CENTRAL} defines three types of entities for security management: users, groups, and roles. You can assign permissions to both roles and groups. Users inherit permissions from the groups and roles that the user is a member of.
```
...roles that they are a member of...
or
A user inherits permissions from the groups and roles that the user is a member of.

kaldesai replied:
Done

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-provider-con.adoc:
```diff
@@ -0,0 +1,11 @@
+
+= Security management providers
+
+In the context of security management, a realm restricts access to different application resources. Realms contain information about the users, groups, roles, and permissions. A concrete user and group management service implementation for a specific realm is called a security management provider.
```
Realms contain information about the users, groups, roles, and permissions.

Remove 'the'.

kaldesai replied:
Done

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-provider-con.adoc:
```diff
@@ -0,0 +1,11 @@
+
+If the built-in security management providers do not meet the requirements of your application security realm, then you can build and register your own security management provider.
+
+NOTE: If the security management provider is not installed, the user interface for managing the security realm will not be available. After you install and configure a security management provider, the user and group management features are automatically enabled in the security management user interface.
```
NOTE: If a security management provider is not installed, the user interface for managing the security realm is not available.

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/configuring-EAP-security-management-provider-CLI-mode-proc.adoc:
```diff
@@ -0,0 +1,44 @@
+
+.Procedure
+
+. To use an existing users or roles property files from the {EAP} instance, include the following system properties in the `EAP_HOME/standalone/configuration/application-users.properties` and `EAP_HOME/standalone/configuration/application-roles.properties` files:
```
property file

Also, I would like to see an example. Show me how these properties will look in the files, then you can follow with the table. This isn't a show stopper but would be nice to do.

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/configuring-EAP-security-management-provider-CLI-mode-proc.adoc:
```diff
@@ -0,0 +1,44 @@
+----
+src/main/resources/security-management.properties
+----
+. Enter the following keys and provider name as a value:
```
How do I do this? On the command line or is this something that I add to the file? Can you give an example?

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/configuring-EAP-security-management-provider-property-files-proc.adoc:
```diff
@@ -0,0 +1,39 @@
+
+.Procedure
+
+. To use an existing users or roles property files from the {EAP} instance, include the following system properties in the `EAP_HOME/standalone/configuration/application-users.properties` and `EAP_HOME/standalone/configuration/application-roles.properties` files:
```
AB$

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/configuring-EAP-security-management-provider-CLI-mode-proc.adoc:
```diff
@@ -0,0 +1,44 @@
+
+|===
+
+. Create the `security-management.properties` file in the root of your application. For example, create the following file:
```
root directory of your...

kaldesai replied:
Done

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/configuring-EAP-security-management-provider-property-files-proc.adoc:
```diff
@@ -0,0 +1,39 @@
+|`org.uberfire.ext.security.management.wildfly.properties.groups-file-path` | The absolute file path for the groups property file. This property is mandatory. |`./standalone/configuration/application-roles.properties`
+
+|===
+. Create the `security-management.properties` file in the root of your application. For example, create the following file:
```
root directory of your...

kaldesai replied:
Done

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-security-management-permissions-con.adoc:
```diff
@@ -0,0 +1,12 @@
+* View a repository.
+* Delete a dashboard.
+
+You can grant or deny permission and a permission can be global or resource specific. You can use permissions to protect access to resources and customize features within the application.
```
deny a permission

kaldesai replied:
Done

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-changing-permissions-proc.adoc:
```diff
@@ -0,0 +1,31 @@
+[id='business-central-settings-changing-permissions-proc_{context}']
+= Changing permissions for groups and roles in {CENTRAL}
+
+In {CENTRAL}, you cannot change permissions for an individual user. However, you can change permissions for groups and roles. The changed permissions apply to users that have the role or belong to the group that you changed.
```
The changed permissions apply to users with the role or that belong to a group that you changed.

kaldesai replied:
Done

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-changing-permissions-proc.adoc:
```diff
@@ -0,0 +1,31 @@
+
+[NOTE]
+====
+Any changes you make to roles or groups affect all of the users associated with that role or group.
```
Any changes that you make...

kaldesai replied:
Done

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-changing-permissions-proc.adoc:
```diff
@@ -0,0 +1,31 @@
+Any changes you make to roles or groups affect all of the users associated with that role or group.
+====
+
+.Procedure
```
Any prereqs? You are logged in to BC? Do I have to have the admin role?

kaldesai replied:
Added the Prerequisites.

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-changing-permissions-proc.adoc:
```diff
@@ -0,0 +1,31 @@
+.Setting the permissions
+image::Workbench/SecurityManagement/set-permissions.png[Setting the permissions]
++
+You can expand the resource type whose permissions you want to change.
```
You can expand the resource type whose permissions you want to change.
Change to:
6. Click the arrow next to a resource type to (and I'm not sure just what happens when you do this or why you would want to...)

kaldesai replied:
When you click the arrow next to a resource type it expands the resource type and you can change the permission by using the slider button.

kaldesai commented:
Hey @emmurphy1, I have updated the content as per your suggestions in the PR. Can you please take a look once again? Thank you.

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/configuring-EAP-security-management-provider-CLI-mode-proc.adoc:
```diff
@@ -0,0 +1,52 @@
+
+= Configuring the {EAP} security management provider based on property files and CLI mode
+
+To use the {EAP} security management provider based on property files and CLI mode, complete the following steps:
```
If the steps immediately followed this statement, the : would be correct. I suggest changing to:
...complete the steps in this procedure.

emmurphy1 replied:
Kalyani, I don't see this change in the rendered draft, or here.

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/configuring-EAP-security-management-provider-CLI-mode-proc.adoc:
```diff
@@ -0,0 +1,44 @@
+.Procedure
+
+. To use an existing users or roles property files from the {EAP} instance, include the following system properties in the `EAP_HOME/standalone/configuration/application-users.properties` and `EAP_HOME/standalone/configuration/application-roles.properties` files:
++
```
Add an introduction to the table.
The following table provides a description and default value for these properties:

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-changing-home-proc.adoc:
```diff
@@ -1,7 +1,7 @@
+= Changing the {CENTRAL} home page
 
-The home page is the page that appears after you log on to {CENTRAL}. By default, the home page is set to Home. You can change the home page for roles and groups.
+The home page is the page that appears after you log in to {CENTRAL}. By default, the home page is set to Home. You can specify a different home page for each role and group.
```
set to Home. Home should be in code.

emmurphy1 commented on doc-content/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/business-central-settings-setting-priorities-proc.adoc:
```diff
@@ -1,4 +1,4 @@
+[id='business-central-settings-setting-priorities-proc_{context}']
 = Setting priorities
 
 A user can have multiple roles and belong to multiple groups. The Priority setting determines the order of precedence of a role or group.
```
Any prerequisties? Like You are logged in to BC with a user that has the admin role?
